ext/dom/nodelist.c

```diff
@@ -31,6 +31,23 @@
 * Since:
 */
 
+static zend_always_inline void objmap_cache_release_cached_obj(dom_nnodemap_object *objmap)
+{
+	if (objmap->cached_obj) {
+		if (GC_DELREF(&objmap->cached_obj->std) == 0) {
+			zend_objects_store_del(&objmap->cached_obj->std);
+		}
+		objmap->cached_obj = NULL;
+		objmap->cached_obj_index = 0;
+	}
```

Comment:
Just to confirm, this is done manually as we are introducing a cyclic dependency on the object, right?

Reply:
No. I needed to take an additional reference to the dom_object, because now also cached_obj refers to the same dom_object that was returned by the last iteration. I could do this via a zval, but since a zval is 16 bytes and a pointer is 8 bytes, using a pointer is more lightweight.
I just manage the refcount manually like this instead of using a zval and its APIs. This requires me to also manually free it. I don't have to account for cycles because the DOM is a tree, so it can't have cycles by definition.

Reply:
Right, okay makes sense!

Could you just write this up in a comment, so the next person trying to understand ext-dom has an easier time?